app/request/context.go

```diff
@@ -116,3 +119,27 @@ func (c *Context) SetAppError(err *model.AppError) {
 func (c *Context) AppError() *model.AppError {
 	return c.err
 }
+
+type CTX interface {
```

Comment:
`CTX` in all-caps looks a bit odd. Can we use something else? Only 1/5 though.

Reply:
Ok, let's keep it. No worries. :+1:

Reply:
IMO, there is no harm on long interface names. Eg. `ContextInterface` wouldn't hurt. 1/5 :p

Reply:
`ContextInterface` sounds very Java like :see_no_evil:

Reply:
And instead of `EmptyContext` we need to name it `ContextFactoryImpl` ;-)